spec/envelope (#76)
vacp2p/rfc: edf.md

# Envelope data field

> Version: 0.1.0 (Draft)
> 
> Authors: Oskar Thorén <oskar@status.im>, Dean Eigenmann <dean@status.im>

> ## Table of Contents

Reviewer: Can come in a follow up PR (issue) I suppose, but would be a good idea to be more clear about design goals here. E.g. see something like https://code.briarproject.org/briar/briar-spec/blob/master/protocols/BTP.md re design requirements, adversary model, underlying, pre-reqs.

Author: @oskarth does that really belong in here?

Reviewer: What's the motivation for the spec then? If it can't promise any guarantees under conditions or state requirements, then what is it?

 1. [Abstract](#abstract)
 2. [Specification](#specification)
    1. [ABNF](#abnf)
    2. [Signature](#signature)
    3. [Padding](#padding)
 3. [Copyright](#copyright)

## Abstract

> In this specification, we describe the [data field used in Waku](./waku.md#abnf-specification), this is relevant for decrypting incoming messages.

Reviewer: might be worth reffing to eip627 as we took it from there

Reviewer 2: Come to think if it, it is a bit of a weird sentence in the context of this separate spec.

Maybe change into:
```suggestion
In this specification, we describe the [data field used in Waku](./waku.md#abnf-specification) for the encryption, decryption and signing of messages.
```

And for clarification, we could add something in the waku spec as:
Understanding the format of the data field of envelopes is only required when posting new envelopes or decrypting incoming envelopes, not when a node is only forwarding envelopes.

# Envelope data field

> Version: 0.1.0 (Draft)
> 
> Authors: Oskar Thorén <oskar@status.im>, Dean Eigenmann <dean@status.im>

## Table of Contents

 1. [Abstract](#abstract)
 2. [Specification](#specification)
    1. [ABNF](#abnf)
    2. [Signature](#signature)
    3. [Padding](#padding)
 3. [Copyright](#copyright)

## Abstract

In this specification, we describe the [data field used in Waku](./waku.md#abnf-specification), this is relevant for decrypting incoming messages.

## Specification

The `data` field is used within the `waku envelope`, the field MUST contain the encrypted message of the envelope.

<!--

> do we care about this section?

Reviewer 2: It can go I think, considering the abstract.
